OpenRA.Mods.Common/Traits/DockClient.cs

```diff
@@ -0,0 +1,136 @@
+﻿#region Copyright & License Information
+/*
+ * Dock client module by Boolbada of OP Mod.
+ *
+ * OpenRA Copyright info:
+ *
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class DockClientInfo : ITraitInfo
+	{
+		public object Create(ActorInitializer init) { return new DockClient(init, this); }
+	}
+
+	public enum DockState
+	{
+		NotAssigned,
+		WaitAssigned,
+		ServiceAssigned
+	}
+
+	// When dockmanager manages docked units, these units require dock client trait.
+	public class DockClient : INotifyKilled, INotifyBecomingIdle, INotifyActorDisposing, INotifyOwnerChanged, IResolveOrder
```

Comment:
Is there a specific reason for not releasing on `INotifyRemovedFromWorld`?  That then replaces the need for `INotifyKilled`, `INotifyActorDisposing`, `INotifyOwnerChanged`, and covers additional cases such as being removed from the world using Lua or transports.

Reply:
Switched to INotifyRemovedFromWorld. I thought it is triggered when the actor enters the transport and gets removed from the world.